pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -441,19 +439,45 @@ func (r *ReconcileWindowsMachine) isAllowedDeletion(machine *mapi.Machine) bool
 		return false
 	}
 
-	if minHealthyCount == *(windowsMachineSet.Spec.Replicas) {
+	// Allow deletion if there is only one machine in the Windows MachineSet
+	totalWindowsMachineCount := *windowsMachineSet.Spec.Replicas
+	if maxUnhealthyCount == totalWindowsMachineCount {
 		return true
 	}
 
 	totalHealthy := 0
 	for _, ma := range machines.Items {
-		// Machines are determined as Healthy when they are part of Windows MachineSet and are
-		// in the Running Status
-		if len(ma.OwnerReferences) != 0 && ma.OwnerReferences[0].Name == machinesetName {
-			if ma.Status.Phase != nil && *ma.Status.Phase == "Running" && ma.Status.NodeRef != nil {
-				totalHealthy += 1
-			}
+		// Increment the count if the machine is identified as healthy and is a part of given Windows MachineSet and
+		// on which deletion is not already initiated.
+		if len(machine.OwnerReferences) != 0 && ma.OwnerReferences[0].Name == machinesetName &&
```

Comment:
nit: The len(machine.OwnerReferences) != 0 check is already made above https://github.com/openshift/windows-machine-config-operator/pull/148/files#diff-f6cf75653deff7ed72be94f059acc292d15113ed7c9f9ce0fdc5e842fa71aa72R422